tests/pytorch_tests/model_tests/feature_models/matmul_test.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2024 Sony Semiconductor Israel, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import torch
+from tests.pytorch_tests.model_tests.base_pytorch_test import BasePytorchTest
+
+"""
+This test checks the MatMul substitution function.
+"""
+
+
+class MatMulFNet(torch.nn.Module):
+    """
+    Model for testing MatMul function
+    """
+    def __init__(self):
+        super(MatMulFNet, self).__init__()
+
+    def forward(self, x):
+        x_t = torch.transpose(x, -1, -2)
```

Comment:
This doesn't test inputs with different number of dimensions.
For example: need to test `(B, d1, d2, d3, a, b) @ (B, a, b)`, `(B, a, b) @ (B, d1, d2, a, b)`